### PR TITLE
Update SRG-OS-000312-GPOS-00123

### PR DESCRIPTION
**Pull request by Mab879**
#### Description:

Update SRG-OS-000312-GPOS-00123 for RHEL9 STIG.

See the commits for details.

#### Rationale:
RHEL9 STIG

### CODE REVIEW

File: controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml [hunk 1]
```diff
@@ -5,5 +5,6 @@ controls:
         title: {{{ full_name }}} must allow operating system admins to grant their privileges
             to other operating system admins.
         rules:
-            - use_pam_wheel_for_su
+            - sysctl_fs_protected_symlinks
+            - sysctl_fs_protected_hardlinks
```
Comment (yuumasato) [inline controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml:9]: Given the control title, it seems to me that `use_pam_wheel_for_su` makes more sense  than the rules added.
https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/accounts/accounts-restrictions/root_logins/use_pam_wheel_for_su/rule.yml#L30

**Review by yuumasato — CHANGES_REQUESTED** [review]

**Comment by yuumasato** [comment]
@Mab879 Just to understand, the SRGs are being added to the rule because RHEL8 uses them?

**Comment by Mab879** [comment]
> @Mab879 Just to understand, the SRGs are being added to the rule because RHEL8 uses them?

That is correct.

File: controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml (hunk 1 above)
Comment (yuumasato) [reply controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml:10]: Could you clarify the reasoning? To me the symlink rules don't make sense to this SRG.

Comment (yuumasato) [reply controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml:10]: When the controls have value 0, they symlink behavior is unrestricted  (https://sysctl-explorer.net/fs/protected_hardlinks/ https://sysctl-explorer.net/fs/protected_symlinks/)

I see the symlink syscontrols as a restriction mechanism rather then a "granting" or delegation mechanism, so I fail to see the connection.

If you have a strongly we can merge this, but I think we should avoid adding rules to SRGs with the link  /rationale  is weak.

Comment (Mab879) [reply controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml:10]: According to the [latest RHEL8 STIG](https://stigs.mab879.com/stigs/5/controls/RHEL-08-010373) these rules satisfy SRG-OS-000312-GPOS-00122, SRG-OS-000312-GPOS-00123, SRG-OS-000312-GPOS-00124, SRG-OS-000324-GPOS-00125.

I was adding trying to match the RHEL8 STIG.

Comment (yuumasato) [reply controls/srg_gpos/SRG-OS-000312-GPOS-00123.yml:10]: Let's move forward with this, and we can change it later.

**Review by yuumasato — APPROVED** [review]

**Pull request merged**